qiskit/opflow/gradients/circuit_gradients/lin_comb.py

```diff
@@ -665,11 +692,45 @@ def _gradient_states(
                         param_expression = gate.params[idx]
 
                         if isinstance(meas_op, OperatorBase):
-                            state = meas_op @ state
-                        elif meas_op is True:
-                            state = ListOp(
-                                [state], combo_fn=partial(self._grad_combo_fn, state_op=state_op)
-                            )
+                            state = StateFn(meas_op, is_measurement=True) @ state
+
+                        else:
+                            if aux_meas_op == Z - 1j * Y:
+                                state_z = ListOp(
+                                    [state],
+                                    combo_fn=partial(self._grad_combo_fn, state_op=state_op),
+                                )
+                                pbc = PauliBasisChange(
+                                    replacement_fn=PauliBasisChange.measurement_replacement_fn
+                                )
+                                pbc = pbc.convert(-Y ^ (I ^ (state.num_qubits - 1)))
+                                state_y = pbc[-1] @ state
+                                state_y = ListOp(
+                                    [state_y],
+                                    combo_fn=partial(self._grad_combo_fn, state_op=state_op),
+                                )
+                                state = state_z - 1j * state_y
+
+                            elif aux_meas_op == -Y:
+                                pbc = PauliBasisChange(
+                                    replacement_fn=PauliBasisChange.measurement_replacement_fn
+                                )
+                                pbc = pbc.convert(aux_meas_op ^ (I ^ (state.num_qubits - 1)))
+                                state = pbc[-1] @ state
+                                state = -1 * ListOp(
+                                    [state],
+                                    combo_fn=partial(self._grad_combo_fn, state_op=state_op),
+                                )
+                            elif aux_meas_op == Z:
+                                state = ListOp(
+                                    [state],
+                                    combo_fn=partial(self._grad_combo_fn, state_op=state_op),
+                                )
+                            else:
+                                raise ValueError(
+                                    "The decomposition of the auxiliary measurement "
+                                    "operator went wrong."
```

Comment:
I think we could be a bit more specific here:
```suggestion
                                    f"The auxiliary measurement operator passed {aux_meas_op}"
                                    "is not supported. Only Y, Z, or Z - 1j * Y are valid."
```